src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3192,79 +3191,52 @@ private IOperation CallNullableMember(IOperation value, SpecialMember nullableMe
 
             if (!isOnStatementLevel)
             {
-                frames.Push(PushStackFrame());
+                expressionFrame = PushStackFrame();
             }
 
             IConditionalAccessOperation currentConditionalAccess = operation;
 
-            while (true)
-            {
-                frames.Push(PushStackFrame());
-
-                if (currentConditionalAccess.WhenNotNull.Kind != OperationKind.ConditionalAccess)
-                {
-                    break;
-                }
-
-                currentConditionalAccess = (IConditionalAccessOperation)currentConditionalAccess.WhenNotNull;
-            }
-
-            var whenNull = new BasicBlockBuilder(BasicBlockKind.Block);
-
             currentConditionalAccess = operation;
             IOperation testExpression;
+            var whenNull = new BasicBlockBuilder(BasicBlockKind.Block);
+            var previousTracker = _currentConditionalAccessTracker;
+            _currentConditionalAccessTracker = new ConditionalAccessOperationTracker(operations, whenNull);
 
             while (true)
             {
-                testExpression = currentConditionalAccess.Operation;
-                SyntaxNode testExpressionSyntax = testExpression.Syntax;
-                ITypeSymbol? testExpressionType = testExpression.Type;
-
-                PushOperand(VisitRequired(testExpression));
-                SpillEvalStack();
-                IOperation spilledTestExpression = PopOperand();
-                PopStackFrame(frames.Pop());
-
-                ConditionalBranch(MakeIsNullOperation(spilledTestExpression),
-                    jumpIfTrue: true,
-                    whenNull);
-                _currentBasicBlock = null;
-
-                IOperation receiver = OperationCloner.CloneOperation(spilledTestExpression);
-
-                if (ITypeSymbolHelpers.IsNullableType(testExpressionType))
+                if (!IsConditionalAccessInstancePresentInChildren(currentConditionalAccess.WhenNotNull))
                 {
-                    receiver = CallNullableMember(receiver, SpecialMember.System_Nullable_T_GetValueOrDefault);
+                    // https://github.com/dotnet/roslyn/issues/27564: It looks like there is a bug in IOperation tree around XmlMemberAccessExpressionSyntax,
+                    //                      a None operation is created and all children are dropped.
+                    //                      See Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator.UnitTests.ExpressionCompilerTests.ConditionalAccessExpressionType
+                    //                      Because of this, the recursion to visit the child operations will never occur if we visit the WhenNull of the current
+                    //                      conditional access, so we need to manually visit the Operation of the conditional access now.
+                    testExpression = VisitConditionalAccessTestExpression(currentConditionalAccess.Operation);
```

Comment:
>testExpression = VisitConditionalAccessTestExpression(currentConditionalAccess.Operation); [](start = 20, length = 90)

It look like an initialization of  ```testExpression``` with anything other than ```currentConditionalAccess.Operation``` is unexpected. For example, we do the following test below ```operation.Operation == testExpression```.

Reply:
The updates to `ConditionalAccess_11` and `ConditionalAccess_12` will reflect addressing this feedback.

---
In reply to: [573174786](https://github.com/dotnet/roslyn/pull/51032#discussion_r573174786) [](ancestors = 573174786)